pkg/console/operator/status.go

```diff
@@ -0,0 +1,299 @@
+		Status:             operatorsv1.ConditionTrue,
+		Reason:             conditionReason,
+		Message:            conditionMessage,
+		LastTransitionTime: metav1.Now(),
+	})
+
+	return operatorConfig
+}
+
+func (c *consoleOperator) ConditionNotFailing(operatorConfig *operatorsv1.Console) *operatorsv1.Console {
+	v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorsv1.OperatorCondition{
+		Type:               operatorsv1.OperatorStatusTypeFailing,
+		Status:             operatorsv1.ConditionFalse,
+		LastTransitionTime: metav1.Now(),
+	})
+
+	return operatorConfig
+}
+
+func (c *consoleOperator) ConditionProgressing(operatorConfig *operatorsv1.Console) *operatorsv1.Console {
+	v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorsv1.OperatorCondition{
+		Type:               operatorsv1.OperatorStatusTypeProgressing,
+		Status:             operatorsv1.ConditionTrue,
+		LastTransitionTime: metav1.Now(),
+	})
+
+	return operatorConfig
+}
+func (c *consoleOperator) ConditionNotProgressing(operatorConfig *operatorsv1.Console) *operatorsv1.Console {
+	v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorsv1.OperatorCondition{
+		Type:               operatorsv1.OperatorStatusTypeProgressing,
+		Status:             operatorsv1.ConditionFalse,
+		LastTransitionTime: metav1.Now(),
+	})
+
+	return operatorConfig
+}
+
+func (c *consoleOperator) ConditionAvailable(operatorConfig *operatorsv1.Console) *operatorsv1.Console {
+	v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorsv1.OperatorCondition{
+		Type:               operatorsv1.OperatorStatusTypeAvailable,
+		Status:             operatorsv1.ConditionTrue,
+		LastTransitionTime: metav1.Now(),
+	})
+
+	return operatorConfig
+}
+
+func (c *consoleOperator) ConditionNotAvailable(operatorConfig *operatorsv1.Console) *operatorsv1.Console {
+	v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorsv1.OperatorCondition{
+		Type:               operatorsv1.OperatorStatusTypeAvailable,
+		Status:             operatorsv1.ConditionFalse,
+		LastTransitionTime: metav1.Now(),
+	})
+
+	return operatorConfig
+}
+
+// A sync failure has happened.
+// We don't necessarily know the condition of the operand (console),
+// message should be a useful error output.
+// but we do know that the operator is failing to update the operand.
+func (c *consoleOperator) ConditionResourceSyncFailure(operatorConfig *operatorsv1.Console, message string) *operatorsv1.Console {
+	v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorsv1.OperatorCondition{
+		Type:               operatorsv1.OperatorStatusTypeFailing,
+		Status:             operatorsv1.ConditionTrue,
+		Message:            message,
+		Reason:             reasonSyncError,
+		LastTransitionTime: metav1.Now(),
+	})
+
+	return operatorConfig
+}
+
+func (c *consoleOperator) ConditionResourceSyncSuccess(operatorConfig *operatorsv1.Console) *operatorsv1.Console {
```

Comment:
How is this different than `ConditionNotFailing`?

Reply:
It is not at this point.  Initially I was considering putting the rest of [ this logic (sync_v400 line 110) ](https://github.com/openshift/console-operator/pull/142/files#diff-283b3e79dd03261002107f949ad39c38R110) in  this function as it may have been necessary to set more than one status. 

I may be willing to eliminate this wrapper as I believe it won't be a "set multiple statuses" kind of function.